### PR TITLE
Changes for Issue 5844 - Forgotten AnkiWeb e-mail

### LINKED ISSUE #5844: Feature: Forgotten AnkiWeb Email Instructions
## Story

As a
 
* New AnkiDroid user 
* With an existing AnkiWeb account from Anki Desktop
* Who has forgotten my email

I would like to be able to access instructions on how to retrieve my email from the logon screen so I can retrieve my data from AnkiWeb.

----

Ref: https://groups.google.com/forum/#!topic/anki-android/I92p5ta892o

### PR DESCRIPTION
**Pull request by sethbuchinger**
## Purpose / Description
Allow user the ability to click a link for instructions on how to find their AnkiWeb Account Email

## Fixes
Fixes #5844

## Approach
This change adds a button on the login screen for the user to click to open up a web browser with instructions on how to find their AnkiWeb email. 

Screen Shot 2021-11-17 at 6 09 39 PM

## How Has This Been Tested?

This was tested on a Pixel3A_API_31_arm64-v8a emulator with  SDK version: Android 12.0 (S)

## Checklist

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [X] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/MyAccount.java [hunk 1]
```diff
@@ -227,6 +227,12 @@ private void initAllContentViews() {
         Uri url = Uri.parse(getResources().getString(R.string.register_url));
         signUpButton.setOnClickListener(v -> openUrl(url));
 
+        //Add new button to link to instructions on how to find AnkiWeb email
+        Button lostEmail = mLoginToMyAccountView.findViewById(R.id.lost_mail_instructions);
+        Uri lostMailUrl = Uri.parse("https://github.com/ankidroid/Anki-Android/wiki/FAQ#forgotten-ankiweb-email-instructions");
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/MyAccount.java:232]: Could you move this to `constants.xml`

File: AnkiDroid/src/main/java/com/ichi2/anki/MyAccount.java [hunk 2]
```diff
@@ -227,6 +227,12 @@ private void initAllContentViews() {
         Uri url = Uri.parse(getResources().getString(R.string.register_url));
         signUpButton.setOnClickListener(v -> openUrl(url));
 
+        //Add new button to link to instructions on how to find AnkiWeb email
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/MyAccount.java:230]: 'new' is time relative, this comment will hopefully be here for 10 years :)

```suggestion
        // Add button to link to instructions on how to find AnkiWeb email
```

File: AnkiDroid/src/main/res/values/04-network.xml [hunk 3]
```diff
@@ -43,6 +43,7 @@
     <string name="alert_logging_message">Logging in…</string>
     <string name="invalid_username_password">Invalid email address or password</string>
     <string name="reset_password">Reset password</string>
+    <string name="lost_mail_instructions">Forgot AnkiWeb Account Email?</string>
```
Comment (david-allison-1) [inline AnkiDroid/src/main/res/values/04-network.xml:46]: Optional, I'd prefer this to be more concise. Implementer's choice

```suggestion
    <string name="lost_mail_instructions">Forgot Email?</string>
```

**Review by david-allison-1 — CHANGES_REQUESTED** [review]
Looks great (and really concise!), thanks!

Two nitpicks, and one request on moving the constant

File: AnkiDroid/src/main/java/com/ichi2/anki/MyAccount.java (hunk 1 above)
Comment (sethbuchinger) [reply AnkiDroid/src/main/java/com/ichi2/anki/MyAccount.java]: Added to url to `constants.xml`

File: AnkiDroid/src/main/java/com/ichi2/anki/MyAccount.java (hunk 2 above)
Comment (sethbuchinger) [reply AnkiDroid/src/main/java/com/ichi2/anki/MyAccount.java]: Removed "new" in comment

File: AnkiDroid/src/main/res/values/04-network.xml (hunk 3 above)
Comment (sethbuchinger) [reply AnkiDroid/src/main/res/values/04-network.xml]: Implemented as suggested for conciseness

**Review by david-allison-1 — APPROVED** [review]
Awesome, thanks so much!

**Comment by david-allison-1** [comment]
Just to note for future: You can add `Fixes #5844` to the PR description and it will automatically link + close the PR when merged.

I've edited your post to do this.

**Review by mikehardy — APPROVED** [review]
Nice! This looks really good, thanks for posting it

**Comment by mikehardy** [comment]
Hi there! November OpenCollective Notice

Apologies for the delay, and if you're already submitted something to OpenCollective for PRs merged in November 2021, you may feel free to ignore this and it will be processed shortly

Just a friendly notice that we try to process OpenCollective payments monthly - it's time for November 2021 submissions

If you are interested in compensation for this work, the process with details is here:

https://github.com/ankidroid/Anki-Android/wiki/OpenCollective-Payment-Process#how-to-get-paid

(I only post one comment per person to avoid spamming you, regardless of the number of PRs merged, but this note applies to all PRs merged for in the month of August)

Thanks!

**Pull request merged**